lib/mailbox.dart

```diff
@@ -83,15 +84,34 @@ class Mailbox {
     });
   }
 
+  /// Close a mailbox.
+  ///
+  /// If mailbox already contains a message then it will be dropped.
+  void close() => _mutex.runLocked(() {
+        if (_mailbox.ref.state == _stateFull && _mailbox.ref.bufferLength > 0) {
+          malloc.free(_mailbox.ref.buffer);
+        }
+
+        _mailbox.ref.state = _stateClosed;
+        _mailbox.ref.buffer = nullptr;
+        _mailbox.ref.bufferLength = 0;
+
+        _condVar.notify();
+      });
+
   /// Take a message from the mailbox.
   ///
   /// If mailbox is empty this will synchronously block until message
   /// is available.
```

Comment:
Please update documentation here to mention that `StateError` is thrown if sender closes mailbox.